### PR TITLE
ThirdParty: Update Diligent.

### PR DESCRIPTION
**Pull request by eugeneko**
- Push minimum Android SDK version to 28

- Move manual patches from Diligent repo to ThirdParty CMakeLists.txt

- Fix Thread::ThreadFunctionStatic which didn't work with new Android SDK.

- Cherry pick https://github.com/libsdl-org/SDL/commit/4222cebef697cc6f3e0759b05585c4b00d5df65e

### CODE REVIEW

File: Source/ThirdParty/SDL/src/sensor/android/SDL_androidsensor.c [hunk 1]
```diff
@@ -171,7 +171,7 @@ SDL_ANDROID_SensorUpdate(SDL_Sensor *sensor)
     ASensorEvent event;
     struct android_poll_source* source;
 
-    if (ALooper_pollAll(0, NULL, &events, (void**)&source) == LOOPER_ID_USER) {
+    if (ALooper_pollOnce(0, NULL, &events, (void**)&source) == LOOPER_ID_USER) {
```
Comment (gleblebedev) [inline Source/ThirdParty/SDL/src/sensor/android/SDL_androidsensor.c:174]: Is this change related to Diligent?

File: android/build.gradle [hunk 2]
```diff
@@ -25,14 +25,14 @@ android {
...
-        minSdkVersion 17
+        minSdkVersion 28
         targetSdkVersion 33
         ndk {
             abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86_64'
         }
         externalNativeBuild {
             cmake {
-                arguments '-DANDROID_STL=c++_static', '-DANDROID_PLATFORM=android-21', '-DANDROID=1',
+                arguments '-DANDROID_STL=c++_static', '-DANDROID_PLATFORM=android-28', '-DANDROID=1',
```
Comment (gleblebedev) [inline android/build.gradle:35]: This should probably reflected in documentation somewhere. So we going from Android 5.0 (99.6% of devices) to Android 9.0 (91.7% of devices)? Is it absolutely necessary?

Comment (rokups) [reply android/build.gradle:35]: IMO this is really not that important. Users are expected to copy `android` dir and thus `build.gradle` and adapt them to their own needs. They can use any version they like.

File: Source/ThirdParty/CMakeLists.txt [hunk 3]
```diff
@@ -282,14 +282,29 @@ install_third_party_libs (
...
     get_target_property(DILIGENT_COMPILE_DEFS Diligent-PublicBuildSettings INTERFACE_COMPILE_DEFINITIONS)
     string(REPLACE "$<$<CONFIG:DEBUG>:DILIGENT_DEVELOPMENT;DILIGENT_DEBUG>" "" DILIGENT_COMPILE_DEFS "${DILIGENT_COMPILE_DEFS}")
     set_property(TARGET Diligent-PublicBuildSettings PROPERTY INTERFACE_COMPILE_DEFINITIONS ${DILIGENT_COMPILE_DEFS})
 endif ()
+
+# Patch: Diligent does not use install interface, make it explicitly empty
+foreach (TARGET ${DILIGENT_ALL_TARGETS})
+    if (TARGET ${TARGET})
+        get_target_property (DIRS ${TARGET} INTERFACE_INCLUDE_DIRECTORIES)
+        if (NOT "${DIRS}" STREQUAL DIRS-NOTFOUND)
```
Comment (rokups) [inline Source/ThirdParty/CMakeLists.txt:302]: You can simply do `if (DIRS)`

File: Source/Urho3D/Core/Thread.cpp [hunk 4]
```diff
@@ -104,13 +100,7 @@ DWORD WINAPI Thread::ThreadFunctionStatic(void* data)
...
 {
     auto* thread = static_cast<Thread*>(data);
-#if defined(__ANDROID_API__)
-#if __ANDROID_API__ < 26
-    prctl(PR_SET_NAME, thread->name_.c_str(), 0, 0, 0);
-#else
-    pthread_setname_np(pthread_self(), thread->name_.c_str(), thread->name_.Length());
-#endif
-#elif defined(__linux__)
+#if defined(__linux__)
```
Comment (rokups) [inline Source/Urho3D/Core/Thread.cpp:103]: Do we really want to drop support for older android APIs?

**Review by rokups — APPROVED** [review]
Looks good to me.

File: Source/ThirdParty/SDL/src/sensor/android/SDL_androidsensor.c (hunk 1 above)
Comment (eugeneko) [reply Source/ThirdParty/SDL/src/sensor/android/SDL_androidsensor.c:174]: It's related to supporting newer androids, gonna keep it just because why not it's mainstream SDL commit

File: Source/ThirdParty/Diligent/Common/CMakeLists.txt [hunk 5]
```diff
@@ -63,7 +67,7 @@ target_include_directories(Diligent-Common
 PRIVATE
     include
 PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/interface>
+    interface
```
Comment (rokups) [inline Source/ThirdParty/Diligent/Common/CMakeLists.txt:70]: I am a bit surprised this does not cause issues. I thought CMake would complain about local path being in includes when SDK is being created. But i suppose you would have noticed that it is the case, so if nothing screams this must be ok? But idk how/why it would work.

Comment (eugeneko) [reply Source/ThirdParty/Diligent/Common/CMakeLists.txt:70]: I have extracted this patch into our CMakeFiles.txt, it still works exactly like before.
I just don't want to patch 20 of Diligent CMakeFiles
